### PR TITLE
Document the release process

### PR DESCRIPTION
**Pull request by intgr**
This is my vision of how the release process looks like.

[See rendered Markdown here](https://github.com/intgr/django-stubs/blob/document-releases/CONTRIBUTING.md#releasing-django-stubs).

Pinging @flaeppe as the one who has made all the latest releases.

## Related issues

* As promised in https://github.com/typeddjango/django-stubs/pull/2183#issuecomment-2133319819
* DRF: https://github.com/typeddjango/djangorestframework-stubs/pull/661

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -115,3 +115,34 @@ If it does, please add the new generic class to the `_need_generic` list in the
...
+
+    - Increase `version=` value within `setup(...)` call in **both** files. The versions must be in sync.
+    - Update `django-stubs-ext>=>=` dependency in root `setup.py` to the same version number.
+    - Version number `major.minor.patch` is formed as follows:
+
+      `major.minor` version must match newest supported Django release.
+
+      `patch` is sequentially increasing for each stubs release. Reset to `0` if `major.minor` was updated.
+
+    - Use pull request title "Version x.y.z release" by convention.
```
Comment (intgr) [inline CONTRIBUTING.md]: I notice now that flaeppe has been using PR title "Prepare for a release \`5.0.1`". I don't have strong feelings either way. Except for the fact that the "a" there seems ungrammatical and can be omitted.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -115,3 +115,36 @@ If it does, please add the new generic class to the `_need_generic` list in the
...
+   to wait for a second maintainer's approval.
+
+3. A maintainer must [сreate a new GitHub release](https://github.com/typeddjango/django-stubs/releases/new):
+
+    - Under "Choose a tag" enter the new version number. Do **not** use `v` prefix.
+    - Click "Generate release notes".
+    - Look for merged PRs with the ['release notes reminder' label](https://github.com/typeddjango/django-stubs/issues?q=is%3Aopen+is%3Aissue+label%3A%22release+notes+reminder%22)
+      and move them to a separate section at the top, so that they stand out. Remove the label from PRs.
+    - Delete all release notes lines containing `by @pre-commit-ci` or `by @dependabot`, as these
+      are irrelevant for our users.
```
Comment (intgr) [inline CONTRIBUTING.md:146]: Haven't done this for the last few releases. It's an additional manual step, but IMO release notes are too noisy if these are included.

**Review by sobolevn — APPROVED** [review]
+1 from me

**Review by flaeppe — APPROVED** [review]

**Comment by flaeppe** [comment]
Great that we get a release flow documented

**Pull request merged**